Upgrade docsy to v0.13.0

zecakeh opened the pull request:
Smooth transition, got not conflicts.

To compare the changes between this branch and upstream: https://github.com/google/docsy/compare/v0.13.0...zecakeh:matrix-org-docsy:docsy-0.13.0

chalin committed:
Set NPM package version to next unreleased dev version 0.12.1-dev (#2269)

ensag-dev committed:
i18n: adds Occitan locale (#2173)

chalin committed:
Hugo and Netlify NPM pkgs: update & change to peer; update UG (#2274)

chalin committed:
Update package.json: use NPM vers of Node (#2275)

deining committed:
Shortcode 'card': trimming of whitespace (#2278)

chalin committed:
UG cleanup of alert calls: drop color when set to default (#2282)

chalin committed:
A better alert shortcode more markdown friendly (#941)

chalin committed:
Update package.json, Hugo to 0.147.8 (#2286)

chalin committed:
Accessibility color fixes, and falling back to BS defaults for typography and color (#2285)

Co-authored-by: Tiffany Hrabusa <30397949+tiffany76@users.noreply.github.com>

chalin committed:
Add support for BS ScrollSpy to show active TOC entry (#2287)

chalin committed:
Refine scrollspy style and behavior (#2290)

chalin committed:
ScrollSpy: drop smooth scrolling (#2291)

chalin committed:
Refine TOC title styles and add top-of-page link (#2292)

chalin committed:
Show close icon when hamburger menu is open (#2301)

chalin committed:
Hamburger style tweaks to make focused state look better (#2302)

chalin committed:
Always show lang menu in navbar (#2303)

deining committed:
Update Bootstrap to 5.3.7 (#2304)

deining committed:
Update Bootstrap to 5.3.8 (#2316)

chalin committed:
Update dev and peer deps (#2315)

chalin committed:
[UG] Don't cache the sidebar (#2318)

chalin committed:
Upgrade deps, Hugo to 0.150.1 (#2320)

hantang committed:
Update zh-cn.toml in i18n (#2313)

chalin committed:
Link check print pages too (#2322)

chalin committed:
[UG] Fix link to flag img in shortcode section (#2324)

chalin committed:
[UG] Fix links and follow style guide re. drop-down (#2325)

Andygol committed:
Update uk.toml (#2331)

chalin committed:
[feat] Add support for non top-level section pages as roots in the sidebar (#2334)

chalin committed:
Update Hugo to 0.151.0 (#2335)

chalin committed:
Reorg NPM package dependencies, rename site folder to docsy.dev (#2337)

chalin committed:
Update CHANGELOG.md to mention NPM-support fix (#2338)

chalin committed:
Fix left-padding of language menu in a foldable sidebar (#2339)

chalin committed:
Rewrite `sidebar_root:true` as `sidebar_root_for:children` (#2340)

chalin committed:
Add support for "self" sidebar roots (#2341)

chalin committed:
Fix sidebar entry HTML attribute spacing (#2342)

chalin committed:
[UX] Prevent flash when in dark mode (#2343)

chalin committed:
[CI] Script to add build ID to the version (#2344)

chalin committed:
Upgrade Hugo to 0.151.2, and Prettify blog posts (#2348)

chalin committed:
[CI] Update Node LTS version, and Hugo to 0.152.2 (#2350)

chalin committed:
[CI] Update release process instructions, add build-ID helpers and more (#2351)

chalin committed:
[UX] Prevent flash when in dark mode, part two (#2355)

chalin committed:
[blog] Upgrading to Docsy 0.12.0 from 0.11.0 (#2349)

chalin committed:
[blog] Adjust titles and slugs for consistency (#2356)

chalin committed:
Don't warn when root section declares `sidebar_root_for:children` (#2361)

chalin committed:
[CI] Rework fix:version so it can be used in a pre-commit hook (#2362)

chalin committed:
Sidebar-root feature & 0.13-wrapup planning and tasks, and more (#2363)

chalin committed:
[EXPERIMENTAL] Sidebar-root UG and CHANGELOG additions (#2364)

chalin committed:
Adjust ScrollSpy TOC tracking and add docs, and more (#2366)

chalin committed:
[CI] Update build & release related scripts and contrib docs (#2367)

chalin committed:
UG adjustments to existing content, and task-file updates pre-0.13.0 (#2369)

chalin committed:
Implement a workaround to ScrollSpy's heading-ID problem (#2370)

chalin committed:
Add _param.html shortcode (#2371)

chalin committed:
fix: ensure ScrollSpy can always find the TOC (#2373)

chalin committed:
chore: refcache refresh and 404 fixes (#2374)

chalin committed:
chore: refcache refresh and more (#2375)

chalin committed:
fix: TOC entry colors in dark mode (#2376)

chalin committed:
fix: TOC entry color contrast, even better (#2379)

chalin committed:
Add color adjustments for dark mode - EXPERIMENTAL (#2380)

chalin committed:
Apply dark-mode color adjustments to the UG (#2381)

chalin committed:
[UG] Navigation page: switch to navbar & sidenav terminology, cleanup markdown (#2372)

chalin committed:
Add ScrollSpy patch workflow and runtime patch (#2382)

chalin committed:
Remove ScrollSpy safeIds workaround and related docs (#2383)

chalin committed:
Refactor dark mode config, add experimental style-only option (#2384)

chalin committed:
Fix spacing of page meta re. dark-mode (#2385)

chalin committed:
Add dark mode support for Google search (#2387)

chalin committed:
Site docs reorg in prep for more design and ops docs, also updated tasks/0.13.0 pages (#2388)

chalin committed:
Improved light/dark-mode docs (#2386)

chalin committed:
Add implementation docs and update changelog for 0.13.0 (#2391)

chalin committed:
docs: swap slugs for 'adding-content' and 'content' (#2392)

chalin committed:
Add missing link def to navigation.md, fix linter config, and more (#2393)

deining committed:
KaTeX: drop js script files, use build-time rendering instead (#2276)

Co-authored-by: Patrice Chalin <chalin@noreply.github.com>

chalin committed:
Scripts partial adj. for Katex, drop mhchem (#2394)

chalin committed:
Add note about section-specific passthrough hook usage, fix KaTeX display elt width (#2395)

chalin committed:
Skip version check on main branch in pre-push hook (#2397)

chalin committed:
Update changelogs and reports for KaTeX build-time rendering (#2398)

chalin committed:
Add TOC style customization section to navigation docs (#2399)

chalin committed:
Add i18n support for table of contents labels (#2400)

chalin committed:
Document i18n support for TOC labels, update task docs (#2402)

chalin committed:
Document `notoc`, and update TOC CSS class names (#2405)

chalin committed:
[UX] Show left/right scroll indicators for the navbar on mobile (#2406)

chalin committed:
Update changelog for navbar scroll indicators and notoc docs (#2407)

chalin committed:
Change link title from 'Documentation' to 'Docs' (#2408)

chalin committed:
[Blog] Release 0.13.0 report and upgrade guide (#2368)

chalin committed:
Pre-0.13.0 release adjustments (#2411)

chalin committed:
Release 0.13.0 preparation (#2412)

zecakeh committed:
Merge tag 'v0.13.0' into docsy-0.13.0

zecakeh committed:
Use URL with asset type for markmap-autoloader

richvdh commented:
For reference, an interdiff between the [v0.12 diff](https://github.com/google/docsy/compare/v0.12.0...matrix-org:docsy:5df72519f5af2355348b8cfe4f4cdf320efe37b8) and the [v0.13 diff](https://github.com/google/docsy/compare/v0.13.0...zecakeh:matrix-org-docsy:docsy-0.13.0):

```diff
--- v0.12.0.diff        2026-01-06 18:21:49.884428963 +0000
+++ v0.13.0.diff        2026-01-06 18:23:41.184962495 +0000
@@ -47,10 +47,10 @@
 +  {{ end -}}
 +{{ end -}}
 diff --git c/layouts/_partials/head.html w/layouts/_partials/head.html
-index d25e217..21b5ac9 100644
+index 2f5c24a..40aa8c6 100644
 --- c/layouts/_partials/head.html
 +++ w/layouts/_partials/head.html
-@@ -27,12 +27,12 @@
+@@ -63,12 +63,12 @@
  {{ partial "twitter_cards.html" . -}}
  {{ partialCached "head-css.html" . "head-css-cache-key" -}}
  <script
@@ -65,7 +65,7 @@
    integrity="sha384-203J0SNzyqHby3iU6hzvzltrWi/M41wOP5Gu+BiJMz5nwKykbkUx8Kp7iti0Lpli"
    crossorigin="anonymous"></script>
  {{ end -}}
-@@ -53,7 +53,7 @@
+@@ -89,7 +89,7 @@
  {{ define "algolia/head" -}}

  {{ if and .Site.Params.search (isset .Site.Params.search "algolia") -}}
@@ -75,19 +75,19 @@
    crossorigin="anonymous" />
  {{ end -}}
 diff --git c/layouts/_partials/scripts.html w/layouts/_partials/scripts.html
-index 181af65..aba9683 100644
+index 38f1552..9c96fc1 100644
 --- c/layouts/_partials/scripts.html
 +++ w/layouts/_partials/scripts.html
-@@ -19,7 +19,7 @@
+@@ -18,7 +18,7 @@
    },
  };
  </script>
 -<script src="https://cdn.jsdelivr.net/npm/markmap-autoloader"></script>
-+<script src="{{ partial "dl-asset.html" "https://cdn.jsdelivr.net/npm/markmap-autoloader" }}"></script>
++<script src="{{ partial "dl-asset.html" "https://cdn.jsdelivr.net/npm/markmap-autoloader/dist/index.min.js" }}"></script>
  {{ end -}}

  {{ if .Site.Params.plantuml.enable -}}
-@@ -77,7 +77,7 @@
+@@ -82,7 +82,7 @@
  {{ partial "hooks/body-end.html" . -}}

  {{ define "algolia/scripts" -}}
@@ -125,10 +125,10 @@
 -    <script src="https://cdn.jsdelivr.net/npm/redoc@latest/bundles/redoc.standalone.js"></script>
 +    <script src="{{ partial "dl-asset.html" "https://cdn.jsdelivr.net/npm/redoc@latest/bundles/redoc.standalone.js" }}"></script>
 diff --git c/layouts/swagger/baseof.html w/layouts/swagger/baseof.html
-index b338a58..1b58fe9 100644
+index 8e28dbb..4592987 100644
 --- c/layouts/swagger/baseof.html
 +++ w/layouts/swagger/baseof.html
-@@ -6,7 +6,7 @@
+@@ -8,7 +8,7 @@
    <head>
      {{ partial "head.html" . }}
      <title>{{ if .IsHome }}{{ .Site.Title }}{{ else }}{{ with .Title }}{{ . }} | {{ end }}{{ .Site.Title }}{{ end }}</title>
@@ -137,7 +137,7 @@
    </head>
    <body class="td-{{ .Kind }}">
      <header>
-@@ -27,8 +27,8 @@
+@@ -29,8 +29,8 @@
              {{ if not (.Param "ui.breadcrumb_disable") -}}
                {{ partial "breadcrumb.html" . -}}
              {{ end -}}
@@ -148,17 +148,3 @@
              {{ block "main" . }}{{ end }}
            </main>
          </div>
-diff --git c/userguide/hugo.yaml w/userguide/hugo.yaml
-index 5143fde..3e31070 100644
---- c/userguide/hugo.yaml
-+++ w/userguide/hugo.yaml
-@@ -146,4 +146,9 @@ module:
-     - source: ../CONTRIBUTING.md
-       target: content/project/contributing.md
-
-+security:
-+  http:
-+    mediaTypes:
-+    - ^application/javascript
-+
- # cSpell:ignore docsy github goldmark markmap plantuml readingtime userguide
```

The significant changes are:
 * change to the path for markmap-autoloader, per 4efa6a9d6d030a7d77c1c93e48cab7b4fd19aad3. I'm sure this is fine.
 * removal of the patch to `userguide/hugo.yaml`, which was introduced in 3b5e8146. I guess this is fine too.

richvdh commented:
(@zecakeh if you can provide any information on those two changes, for the archaeological record, that would be useful)

zecakeh commented:
When merging the upstream branch I got a conflict saying that `userguide/hugo.yaml` was removed, so I removed the patch. I also couldn't remember what that patch was for.

Then when testing this I got this error for the markmap-autoloader file in [CI](https://github.com/matrix-org/docsy/actions/runs/20556068336/job/59040390502?pr=7):

```
hugo: collected modules in 10467 msERROR Could not retrieve asset at 'https://cdn.jsdelivr.net/npm/markmap-autoloader' from CDN. Reason: template: _partials/dl-asset.html:11:22: executing "_partials/dl-asset.html" at <resources.GetRemote>: error calling GetRemote: failed to resolve media type for remote resource "https://cdn.jsdelivr.net/npm/markmap-autoloader".
```

It sounds like it couldn't determine the file type, so I updated the URL to point to the same file, but with an extension.

Now that I am thinking more about it, I am wondering if the patch was not another way to fix this same error because `userguide/hugo.yaml` was not actually removed, it was moved to `docsy.dev/hugo.yaml`. Docs for that config field: <https://gohugo.io/configuration/security/#httpmediatypes>.

If they are indeed fixes for the same thing, the advantage of the new fix is that it works for every site using this theme, rather than requiring a workaround in the config of each site.

In any case since we don't use markmap for matrix-spec, this fix will have no effect there.

richvdh commented:
great, thanks.
